File: backend/internal/model/document.go
package model

import (
	"time"

	"github.com/google/uuid"
)

// DocumentStatus representa o status de um documento
type DocumentStatus string

const (
	DocumentStatusProcessing DocumentStatus = "PROCESSING"
	DocumentStatusReady      DocumentStatus = "READY"
	DocumentStatusError      DocumentStatus = "ERROR"
)

// Document representa um documento PDF
type Document struct {
	ID        uuid.UUID      `db:"id"`
	UserID    uuid.UUID      `db:"user_id"`
	FilePath  string         `db:"file_path"`
	Checksum  string         `db:"checksum"`
	Version   int            `db:"version"`
	Status    DocumentStatus `db:"status"`
	PageCount int            `db:"page_count"`
	CreatedAt time.Time      `db:"created_at"`
	UpdatedAt time.Time      `db:"updated_at"`
}

File: frontend/types/document.ts
// Tipos relacionados a documentos PDF

export interface Document {
  id: string
  user_id: string
  file_path: string
  file_url: string
  checksum: string
  version: number
  status: 'uploaded' | 'processing' | 'processed' | 'error'
  page_count: number
  created_at: string
  updated_at: string
}

export interface DocumentListResponse {
  documents: Document[]
  total: number
  limit: number
  offset: number
}

export interface UploadDocumentResponse {
  document: Document
  message: string
}

export interface ProcessDocumentResponse {
  document: Document
  message: string
}
